tests/devapps/Intune-xamarin-Android/PCAWrapper.cs

```diff
@@ -0,0 +1,129 @@
+
+        // redirect URI for the application
+        private static string _redirectURI = $"msauth://com.sameerk.intune.test.xamarin/EHyvOdXj4uLXJXDaOMy5lwANmp0="; // TODO - Replace with your redirectURI
+
+        // TenantID of your organization
+        private static string _tenantID = "7257a09f-53cc-4a91-aca8-0cb6713642a5"; // TODO - Replace with your TenantID. And also replace in the AndroidManifest.xml
+
+        static string[] clientCapabilities = { "ProtApp" }; // It is must to have these capabilities
+
+        // private constructor for singleton
+        private PCAWrapper()
+        {
+            // Create PCA once. Make sure that all the config parameters below are passed
+            // ClientCapabilities - must have ProtApp
+            PCA = PublicClientApplicationBuilder
+                .Create(_clientID)
+                .WithAuthority(_authority)
+                .WithBroker()
+                .WithClientCapabilities(clientCapabilities)
+                .WithTenantId(_tenantID)
+                .WithRedirectUri(_redirectURI)
+                .Build();
+        }
+
+        /// <summary>
+        /// Static constructor to instantiate PCA
+        /// </summary>
+        static PCAWrapper()
+        {
+            Instance = new PCAWrapper();
+        }
+
+        /// <summary>
+        /// Perform the intractive acquistion of the token for the given scope
+        /// </summary>
+        /// <param name="scopes">Scope</param>
+        /// <param name="parentWindow">Parent window</param>
+        /// <returns></returns>
+        internal async Task<AuthenticationResult> DoInteractiveAsync(string[] scopes, Activity parentWindow)
+        {
+            return await PCA.AcquireTokenInteractive(scopes)
+                                    .WithParentActivityOrWindow(parentWindow)
+                                    .WithUseEmbeddedWebView(true)
+                                    .ExecuteAsync()
+                                    .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Acquire the token silently
+        /// </summary>
+        /// <param name="scopes">scopes</param>
+        /// <returns>Authenticaiton result</returns>
+        internal async Task<AuthenticationResult> DoSilentAsync(string[] scopes)
+        {
+            if (PCA == null)
+            {
+                return null;
+            }
+
+            var accts = await PCA.GetAccountsAsync().ConfigureAwait(false);
+            var acct = accts.FirstOrDefault();
+            if (acct != null)
+            {
+                var silentParamBuilder = PCA.AcquireTokenSilent(scopes, acct);
+                var authResult = await silentParamBuilder
+                                            .ExecuteAsync().ConfigureAwait(false);
+                return authResult;
+            }
+            else
+            {
+                throw new MsalUiRequiredException("ErrCode", "ErrMessage");

```

Comment:
Should have better message - Silent call failed in PCAWrapper. No accounts.

Reply:
Removed the redundant check. Leftover from debugging.